interface/repository/pokemon_repository.go

```diff
@@ -0,0 +1,27 @@
+package repository
+
+import (
+	"github.com/wmaldonadoc/academy-go-q42021/domain/model"
+)
+
+type pokemonRepository struct {
+	db []*model.Pokemon
+}
+
+type PokemonRepository interface {
+	FindById(id int) (*model.Pokemon, error)
+}
+
+func NewPokemonRepository(db []*model.Pokemon) PokemonRepository {
+	return &pokemonRepository{db}
+}
+
+func (pr *pokemonRepository) FindById(id int) (*model.Pokemon, error) {
+
+	for _, poke := range pr.db {
+		if poke.ID == id {
+			return poke, nil
+		}
+	}
+	return nil, nil
+}
```

Comment:
Please avoid having unnecessary empty lines, and having return lines separated.

Also, if you are defining your function signature that is going to return an error, please make use of it. 

```suggestion
func (pr *pokemonRepository) FindById(id int) (*model.Pokemon, error) {
	for _, poke := range pr.db {
		if poke.ID == id {
			return poke, nil
		}
	}
	
	return nil, errors.New("Pokemon not found") // or something like this
}
```